test/E2ETest/TyeRunTests.cs

```diff
@@ -138,6 +139,101 @@ public async Task FrontendBackendRunTest()
             }
         }
 
+        [ConditionalFact]
+        [SkipIfDockerNotRunning]
+        public async Task DockerNamedVolumeTest()
+        {
+            using var projectDirectory = CopyTestProjectDirectory("volume-test");
+            var projectFile = new FileInfo(Path.Combine(projectDirectory.DirectoryPath, "tye.yaml"));
+
+            var outputContext = new OutputContext(_sink, Verbosity.Debug);
+            var application = await ApplicationFactory.CreateAsync(outputContext, projectFile);
+
+            // Add a volume
+            var project = ((ProjectServiceBuilder)application.Services[0]);
+            // Remove the existing volume so we can generate a random one for this test to avoid conflicts
+            var volumeName = "tye_docker_volumes_test" + Guid.NewGuid().ToString().Substring(0, 10);
+            project.Volumes.Clear();
+            project.Volumes.Add(new VolumeBuilder(source: null, name: volumeName, "/data"));
+
+            var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (a, b, c, d) => true,
+                AllowAutoRedirect = false
+            };
+
+            var client = new HttpClient(new RetryHandler(handler));
+            var args = new[] { "--docker" };
+
+            await RunHostingApplication(application, args, _sink, async serviceApi =>
+            {
+                var serviceUri = await GetServiceUrl(client, serviceApi, "volume-test");
+
+                Assert.NotNull(serviceUri);
+
+                var response = await client.GetAsync(serviceUri);
+
+                Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+
+                await client.PostAsync(serviceUri, new StringContent("Things saved to the volume!"));
+
+                Assert.Equal("Things saved to the volume!", await client.GetStringAsync(serviceUri));
+            });
+
+            await RunHostingApplication(application, args, _sink, async serviceApi =>
+            {
+                var serviceUri = await GetServiceUrl(client, serviceApi, "volume-test");
+
+                Assert.NotNull(serviceUri);
+
+                // The volume has data persisted
+                Assert.Equal("Things saved to the volume!", await client.GetStringAsync(serviceUri));
+            });
+
+            // Delete the volume
+            await ProcessUtil.RunAsync("docker", $"volume rm {volumeName}");
+        }
+
+        [ConditionalFact]
+        [SkipIfDockerNotRunning]
+        public async Task DockerHostVolumeTest()
```

Comment:
Good thing I wrote a test, `tye run --docker` didn't work on linux 😮

Reply:
We have a bug for that already. Did you fix it?

Reply:
Not yet, the host.docker.internal one was about communication between services in containers. This was because of path normalization.